### PR TITLE
[DNM] Adding Offline support

### PR DESCRIPTION
**Pull request by monty2610**
This PR contain initial changes for converting universal react app to PWA. 
Following changes are done
* created a component for service worker registration
* added next-workbox-webpack-plugin for generating service worker file
* added webpack-pwa-manifest for generating manifest file to make App installable.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,5 @@ docs
 # next.js build output
 .next
 app/.next
+
+static/workbox
```
Comment (vinodloha) [inline .gitignore]: What is the use of it?

File: app/components/templates/Layout/Layout.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ const Layout = ({ children, deviceType, hasRightGutter }: Props): Node => {
           ) : (
             <meta name="viewport" content={`width=${width}`} />
           )}
+          <link rel="manifest" href="/_next/static/manifest.json" />
```
Comment (vinodloha) [inline app/components/templates/Layout/Layout.js]: Follow the same path that we follow for static resources, there is a redirection rule set for static/*

File: app/lib/pwa/options.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+module.exports = {
+  clientsClaim: true,
+  skipWaiting: true,
+  globPatterns: ['.next/static/*', '.next/static/commons/*'],
+  modifyUrlPrefix: {
+    '.next': '/_next',
```
Comment (vinodloha) [inline app/lib/pwa/options.js]: Why is this modifyUrlPrefix needed?

**Comment by monty2610** [comment]
Closing this PR as next-workbox-webpack-plugin is not well maintained and provide less support than next-offline. Raised another PR with next-offline.

**Pull request closed without merging**